connmgr/connmanager.go

```diff
@@ -137,14 +142,24 @@ type Config struct {
 	// to.  If nil, no new connections will be made automatically.
 	GetNewAddress func() (net.Addr, error)
 
-	// Dial connects to the address on the named network. Either Dial or
-	// DialAddr need to be specified (but not both).
+	// Dial connects to the address on the named network.
+	// Only one of Dial, DialAddr, or DialContext can and must be specified.
+	//
+	// Deprecated: Use DialContext instead.
 	Dial func(network, addr string) (net.Conn, error)
 
-	// DialAddr is an alternative to Dial which receives a full net.Addr instead
-	// of just the protocol family and address. Either DialAddr or Dial need
-	// to be specified (but not both).
+	// DialAddr is an alternative to DialContext which receives a full
+	// net.Addr instead of just the protocol family and address.
+	//
+	// Only one of Dial, DialAddr, or DialContext can and must be specified.
+
 	DialAddr func(net.Addr) (net.Conn, error)
+
+	// DialContext connects to the address on the named network using the
+	// provided context.
+	//
+	// Only one of Dial, DialAddr, or DialContext can and must be specified.
+	DialContext func(ctx context.Context, network, addr string) (net.Conn, error)
```

Comment:
For the context to be usable from dcrlnd we either need a `DialContextAddr(context.Context, net.Addr)` variant or this needs to be changed to use net.Addr (and then fix the respective `dcrdDial()` call in the next commit).

Do you think it's reasonable to change this from network+addr to use a net.Addr (since that's strictly more general)?